### ISSUE TITLE
[Bug]: `yarn version` does not work with `main` as default branch

### ISSUE THREAD

**Issue by dcastil** [yarnpkg/berry#3457]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

When I try to execute `yarn version minor`, I get following error:

```
Usage Error: No ancestor could be found between any of HEAD and master, origin/master, upstream/master

$ yarn version [-d,--deferred] [-i,--immediate] <strategy>
```

### To reproduce

1. Clone this repo: http://github.com/dcastil/yarn-version-bug-test
2. Run `yarn version minor` from the repo root

I couldn't use Sherlock because the [Playground link](https://yarnpkg.com/playground) doesn't work.

The reproduction repo is basically just this:

```sh
mkdir yarn-version-bug-test
cd yarn-version-bug-test
yarn init
yarn set version berry
yarn install
yarn set version latest
yarn plugin import version
git init
git checkout -b main
git commit -am "initial commit"
yarn version minor
```

### Environment

```shell
System:
    OS: macOS 11.5.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 14.17.5 - /private/var/folders/9r/_7lflb5d3057t406krm7vhzr0000gn/T/xfs-33741bcc/node
    Yarn: 3.0.2 - /private/var/folders/9r/_7lflb5d3057t406krm7vhzr0000gn/T/xfs-33741bcc/yarn
    npm: 7.21.0 - ~/.npm-global/bin/npm
```

### Additional context

Issue should have been solved already in https://github.com/yarnpkg/berry/issues/3307, but Im still running into the issue with yarn v3.0.2.

**Comment by merceyz** [comment]
Duplicate of https://github.com/yarnpkg/berry/issues/1982

**Comment by merceyz** [comment]
> Issue should have been solved already in #3307, but Im still running into the issue with yarn v3.0.2.

https://github.com/yarnpkg/berry/issues/3307 was for installing git repositories, this is about the version plugin so it's not the same

**Comment by dcastil** [comment]
> #3307 was for about installing git repositories, this is about the version plugin so it's not the same

Oh I didn't even notice. Thanks for the quick info!
